src/irc.py

```diff
@@ -129,52 +126,115 @@ def _install_snap_package(
+    ) -> None:
+        """Configure the service.
 
         Args:
-            path: The path of the file
-            source: The contents of the file to be pushed
+            db: the database configuration
+            matrix: the matrix configuration
+            config: the charm configuration
         """
-        path.write_text(source, encoding="utf-8")
-        logger.info("Pushed file %s", path)
-
-    def _generate_PEM_file_local(self) -> str:
-        """Generate the PEM file content.
+        self._generate_pem_file_local()
+        self._generate_app_registration_local(matrix, config)
+        self._eval_conf_local(db, matrix, config)
+
+    def _generate_pem_file_local(self) -> None:
+        """Generate the PEM file content."""
+        pem_create_command = [
+            "/bin/bash",
+            "-c",
+            f"[[ -f {IRC_BRIDGE_CONFIG_PATH}/irc_passkey.pem ]] || "
+            + "openssl genpkey -out {IRC_BRIDGE_CONFIG_PATH}/irc_passkey.pem "
+            + "-outform PEM -algorithm RSA -pkeyopt rsa_keygen_bits:2048",
+        ]
+        logger.info("Creating PEM file for IRC bridge.")
+        subprocess.run(pem_create_command, shell=True, check=True, capture_output=True)  # nosec
+
+    def _generate_app_registration_local(
+        self, matrix: DatasourceMatrix, config: CharmConfig
+    ) -> None:
+        """Generate the content of the app registration file.
 
-        Returns:
-            A string
+        Args:
+            matrix: the matrix configuration
+            config: the charm configuration
         """
-        pass
-
-    def _generate_conf_local(self) -> str:
+        app_reg_create_command = [
+            "/bin/bash",
+            "-c",
+            f"[[ -f {IRC_BRIDGE_CONFIG_PATH}/appservice-registration-irc.yaml ]] || "
+            f"matrix-appservice-irc -r -f {IRC_BRIDGE_CONFIG_PATH}/appservice-registration-irc.yaml"
+            f" -u http://{matrix.host}:{IRC_BRIDGE_HEALTH_PORT} "
+            f"-c {IRC_BRIDGE_CONFIG_PATH}/config.yaml -l {config.bot_nickname}",
+        ]
+        logger.info("Creating an app registration file for IRC bridge.")
+        subprocess.run(
+            app_reg_create_command, shell=True, check=True, capture_output=True
+        )  # nosec
+
+    def _eval_conf_local(
+        self, db: DatasourcePostgreSQL, matrix: DatasourceMatrix, config: CharmConfig
+    ) -> None:
         """Generate the content of the irc configuration file.
 
-        Returns:
-            A string
+        Args:
+            db: the database configuration
+            matrix: the matrix configuration
+            config: the charm configuration
         """
-        pass
+        with open(f"{IRC_BRIDGE_CONFIG_PATH}/config.yaml", "r", encoding="utf-8") as f:
+            data = yaml.safe_load(f)
+        db_conn = data["database"]["connectionString"]
+        db_string = f"postgres://{db.user}:{db.password}@{db.host}/{db.db}"
```

Comment:
The pydantic model should generate that string, it's a standard URI format for DB access